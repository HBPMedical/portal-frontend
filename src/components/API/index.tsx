export { default as APICore } from './Core';
export { default as APIMining } from './Mining';
export { backendURL, webURL } from './RequestURLS';
