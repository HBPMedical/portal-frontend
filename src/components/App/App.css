body {
  background: url(../../images/body-bg.jpg) top center no-repeat fixed #f5f5f5;
  background-size: 100% auto;
  font-family: 'Open Sans';
  overflow-y: scroll;
}
