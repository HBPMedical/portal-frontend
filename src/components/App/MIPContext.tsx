import { createContext } from 'react';

interface Context {
  showTooltips: boolean;
  toggleTooltip: () => void;
}

const MIPContext = createContext<Context>({
  showTooltips: true,
  toggleTooltip: () => {
    console.log('Starting context');
  }
});

export default MIPContext;
