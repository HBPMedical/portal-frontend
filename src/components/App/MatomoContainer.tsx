import { createInstance, MatomoProvider } from '@datapunt/matomo-tracker-react';
import React from 'react';
import { useGetMatomoQuery } from '../API/GraphQL/queries.generated';

type Props = {};

const MatomoContainer = ({ children }: React.PropsWithChildren<Props>) => {
  const { data: { configuration } = {}, loading } = useGetMatomoQuery();
  const matomoInstance =
    configuration?.matomo &&
    configuration.matomo.enabled &&
    configuration.matomo.urlBase &&
    configuration.matomo.siteId
      ? createInstance({
          urlBase: configuration.matomo.urlBase,
          siteId: parseInt(configuration.matomo.siteId),
          linkTracking: false,
          configurations: {
            disableCookies: true
          }
        })
      : undefined;

  return matomoInstance && !loading ? (
    <MatomoProvider value={matomoInstance}>{children}</MatomoProvider>
  ) : (
    <>{children}</>
  );
};

export default MatomoContainer;
