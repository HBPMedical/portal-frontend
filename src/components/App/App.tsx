import 'bootstrap/dist/css/bootstrap.css';
import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { APICore, APIExperiment, APIMining, APIModel, APIUser } from '../API';
import Article from '../Article/Container';
import ExperimentCreate from '../Create/Container';
import Explore from '../Explore/Container';
import Home from '../Home/Container';
import TOS from '../Home/TOS';
import ExperimentResult from '../Result/Container';
import ExperimentReview from '../Review/Container';
import Footer from '../UI/Footer';
import Galaxy from '../UI/Galaxy';
import Navigation from '../UI/Navigation';
import NotFound from '../UI/NotFound';
import Splash from '../UI/Splash';
import User from '../User/Container';
import './App.css';

import styled from 'styled-components';

export enum InstanceMode {
  Local,
  Federation
}
export interface AppConfig {
  version?: string;
  instanceName?: string;
  mode?: InstanceMode;
  ga?: string;
  galaxyAPIUrl?: string;
  galaxyApacheUrl?: string;
}
interface Props {
  appConfig: AppConfig;
  apiExperiment: APIExperiment;
  apiCore: APICore;
  apiModel: APIModel;
  apiMining: APIMining;
  apiUser: APIUser;
}

const Redirect = ({
  apiUser,
  props
}: {
  apiUser: APIUser;
  props: any;
}): JSX.Element => {
  const {
    location: { search }
  } = props;
  apiUser.login(search).then(() => {
    props.history.push(`/home`);
  });

  return <Splash apiUser={apiUser} {...props} />;
};

const Content = styled.section`
  margin-top: 116px;
`;

const App = ({
  appConfig,
  apiExperiment,
  apiCore,
  apiModel,
  apiMining,
  apiUser
}: Props): JSX.Element => {
  return (
    <div className="App">
      <header>
        <Navigation
          apiExperiment={apiExperiment}
          apiModel={apiModel}
          appConfig={appConfig}
        />
      </header>

      <Content className="main-content">
        <Switch>
          <Route
            path="/"
            exact={true}
            // tslint:disable-next-line jsx-no-lambda
            render={props => <Splash apiUser={apiUser} {...props} />}
          />
          <Route
            path="/services/login/hbp"
            // tslint:disable-next-line jsx-no-lambda
            render={props => <Redirect apiUser={apiUser} props={props} />}
          />
          <Route
            path="/galaxy"
            render={props => <Galaxy appConfig={appConfig} />}
          />
          <Route
            path="/tos"
            // tslint:disable-next-line jsx-no-lambda
            render={props => (
              <TOS
              // apiUser={apiUser}
              // {...props}
              />
            )}
          />
          <Route
            path="/home"
            // tslint:disable-next-line jsx-no-lambda
            render={props => (
              <Home
                apiCore={apiCore}
                apiModel={apiModel}
                apiExperiment={apiExperiment}
                apiUser={apiUser}
                {...props}
              />
            )}
          />
          <Route
            path="/profile"
            // tslint:disable-next-line jsx-no-lambda
            render={props => <User apiUser={apiUser} {...props} />}
          />
          <Route
            path="/explore"
            // tslint:disable-next-line jsx-no-lambda
            render={props => (
              <Explore
                apiCore={apiCore}
                apiMining={apiMining}
                apiModel={apiModel}
                appConfig={appConfig}
                {...props}
              />
            )}
          />
          <Route
            path={['/review/:slug', '/review']}
            // tslint:disable-next-line jsx-no-lambda
            render={props => (
              <ExperimentReview
                apiMining={apiMining}
                apiModel={apiModel}
                apiCore={apiCore}
                appConfig={appConfig}
                {...props}
              />
            )}
          />
          <Route
            path="/experiment/:slug/:uuid"
            // tslint:disable-next-line jsx-no-lambda
            render={() => (
              <ExperimentResult
                apiExperiment={apiExperiment}
                apiModel={apiModel}
                apiCore={apiCore}
              />
            )}
          />
          <Route
            exact={true}
            path="/experiment/:slug"
            // tslint:disable-next-line jsx-no-lambda
            render={() => (
              <ExperimentCreate
                apiExperiment={apiExperiment}
                apiCore={apiCore}
                apiModel={apiModel}
                appConfig={appConfig}
              />
            )}
          />
          <Route
            path={['/articles/:slug', '/articles']}
            // tslint:disable-next-line jsx-no-lambda
            render={props => <Article apiCore={apiCore} {...props} />}
          />

          <Route component={NotFound} />
        </Switch>
      </Content>
      <footer id="footer">
        <Footer appConfig={appConfig} />
      </footer>
    </div>
  );
};

export default App;
