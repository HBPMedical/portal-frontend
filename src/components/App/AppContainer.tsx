import { useReactiveVar } from '@apollo/client';
import React, { useEffect, useState } from 'react';
import { Router } from 'react-router-dom';
import { webURL } from '../API';
import { appConfigVar } from '../API/GraphQL/cache';
import { AppConfig, history } from '../utils';
import App from './App';
import MIPContext from './MIPContext';

const AppContainer = (): JSX.Element => {
  const [showTutorial, setShowTutorial] = useState<boolean>(true);
  const appConfig = useReactiveVar(appConfigVar);

  const toggleTutorial = (): void => {
    localStorage.setItem('seenTutorial', 'true');
    setShowTutorial(!showTutorial);
  };

  useEffect(() => {
    const seenTutorial = localStorage.getItem('seenTutorial') === 'true';

    setShowTutorial(!seenTutorial);

    const fetchData = async () => {
      // Conf written by dockerize
      const response = await fetch(`${webURL}/static/config.json`);
      const data: AppConfig = await response.json();

      appConfigVar(data);
    };

    fetchData().catch(() => {
      appConfigVar({
        instanceName: 'MIP Development',
        version: 'dev',
        datacatalogueUrl: undefined,
        experimentsListRefresh: `${1000 * 60 * 15}`,
        matomo: {
          enabled: false
        }
      });
    });
  }, []);

  return (
    <MIPContext.Provider
      value={{
        showTutorial: showTutorial,
        toggleTutorial
      }}
    >
      <Router history={history}>
        <App appConfig={appConfig} showTutorial={showTutorial} />
      </Router>
    </MIPContext.Provider>
  );
};

export default AppContainer;
