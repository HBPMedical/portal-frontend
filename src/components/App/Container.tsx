import * as React from 'react';
import ReactGA from 'react-ga';
import { Route, Router } from 'react-router-dom';
import { Provider, Subscribe } from 'unstated';

// import UNSTATED from 'unstated-debug';
import {
  APICore,
  APIExperiment,
  APIMining,
  APIModel,
  APIUser,
  webURL
} from '../API'; // as interfaces
import config from '../API/RequestHeaders';
import App, { AppConfig } from '../App/App';
import Splash from '../UI/Splash';
import { history } from '../utils';
import MIPContext from './MIPContext';

// UNSTATED.logStateChanges = process.env.NODE_ENV === "development";

interface State {
  appConfig: AppConfig;
  showTooltip: boolean;
}

class AppContainer extends React.Component<any, State> {
  public state: State = { appConfig: {}, showTooltip: true };
  private apiExperiment = new APIExperiment(config);
  private apiModel = new APIModel(config);
  private apiCore = new APICore(config);
  private apiMining = new APIMining(config);
  private apiUser = new APIUser(config);

  private intervalId: any; // FIXME: NodeJS.Timer | undefined;

  public async componentDidMount(): Promise<
    [void, void, void, void, void, void] | void
  > {
    // Conf written by dockerize
    let appConfig: AppConfig;
    const response = await fetch(`${webURL}/static/config.json`);
    try {
      const config = await response.json();
      appConfig = {
        ...config,
        datacatalogueUrl:
          config.datacatalogueUrl === '0' ? undefined : config.datacatalogueUrl
      };
      this.setState({ appConfig });

      if (appConfig.ga) {
        ReactGA.initialize(appConfig.ga);
      }
    } catch (e) {
      appConfig = {
        instanceName: 'MIP DEV',
        version: 'alpha',
        datacatalogueUrl: undefined
      };

      this.setState({ appConfig });
    }

    await this.apiUser.user();
    if (this.apiUser.state.authenticated) {
      const username =
        this.apiUser.state.user && this.apiUser.state.user.username;

      // Experiments polling and auth by interval
      this.intervalId = setInterval(() => {
        this.apiUser.user().then(() => {
          if (this.apiUser.state.authenticated) {
            this.apiExperiment.all();
          } else {
            clearInterval(this.intervalId);
          }
        });
      }, 10 * 1000);

      return await Promise.all([
        this.apiUser.profile({ username }),
        this.apiExperiment.all(),
        this.apiCore.fetchPathologies(),
        this.apiCore.algorithms(),
        this.apiCore.stats(),
        this.apiModel.all()
      ]);
    }

    return Promise.resolve();
  }

  public componentWillUnmount(): void {
    clearInterval(this.intervalId);
  }

  public render(): JSX.Element {
    const toggleTooltip = (): void => {
      this.setState(state => ({
        ...state,
        showTooltip: !state.showTooltip
      }));
    };

    return (
      <MIPContext.Provider
        value={{ showTooltips: this.state.showTooltip, toggleTooltip }}
      >
        <Provider
          inject={[
            this.apiExperiment,
            this.apiCore,
            this.apiModel,
            this.apiMining,
            this.apiUser
          ]}
        >
          <Router history={history}>
            <Subscribe
              to={[APIExperiment, APICore, APIModel, APIMining, APIUser]}
            >
              {(
                apiExperiment: APIExperiment,
                apiCore: APICore,
                apiModel: APIModel,
                apiMining: APIMining,
                apiUser: APIUser
              ): JSX.Element => {
                const loading = apiUser.state.loading;
                const authenticated = apiUser.state.authenticated;

                return (
                  <>
                    <Route
                      // Callback from the auth server, redirected to the API
                      path="/services/login/hbp"
                      // tslint:disable-next-line jsx-no-lambda
                      render={(props): JSX.Element => {
                        const {
                          location: { search }
                        } = props;
                        apiUser.login(search);

                        return <div />;
                      }}
                    />
                    {!loading && !authenticated && (
                      <Splash
                        forbidden={this.apiUser.state.forbidden}
                        logout={apiUser.logout}
                      />
                    )}
                    {!loading && authenticated && (
                      <App
                        appConfig={this.state.appConfig}
                        apiExperiment={apiExperiment}
                        apiCore={apiCore}
                        apiModel={apiModel}
                        apiMining={apiMining}
                        apiUser={apiUser}
                      />
                    )}
                  </>
                );
              }}
            </Subscribe>
          </Router>
        </Provider>
      </MIPContext.Provider>
    );
  }
}

export default AppContainer;
