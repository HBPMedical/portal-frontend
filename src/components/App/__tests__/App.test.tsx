import { APICore, APIExperiment, APIMining, APIModel } from '../../API';
import config from '../../API/RequestHeaders';
import App from '../../App/App';
import AppContainer from '../../App/Container';
import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

jest.mock('request-promise-native');

describe.skip('Test App components', () => {
  const apiExperiment = new APIExperiment(config);
  const apiModel = new APIModel(config);
  const apiCore = new APICore(config);
  const apiMining = new APIMining(config);

  let props;

  beforeEach(() => {
    props = {
      appConfig: { appConfig: 'MIP' },
      apiExperiment,
      apiCore,
      apiModel,
      apiMining
    };
  });

  it('AppContainer dom renders correctly', () => {
    const wrapper = shallow(<AppContainer />);
    expect(wrapper).toBeDefined();
  });

  it('AppContainer snapshot renders correctly', () => {
    const tree = renderer.create(<AppContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('App dom renders correctly', () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper).toBeDefined();
  });

  it('App snapshot renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <App {...props} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
