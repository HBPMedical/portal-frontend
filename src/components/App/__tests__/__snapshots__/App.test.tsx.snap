// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App components App snapshot renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <header
      className="navbar navbar-default navbar-fixed-top "
      id="topnav"
    >
      <div
        className="container-fluid"
      >
        <div
          className="logo-area"
        >
          <a
            aria-current={null}
            className="navbar-brand"
            href="/home"
            onClick={[Function]}
          >
            Human Brain Project
          </a>
        </div>
        <div
          className="app-instance-name"
        >
          <h2>
            MIP
          </h2>
        </div>
        <ul
          className="nav navbar-nav toolbar pull-right"
        >
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              aria-current={null}
              href="/explore"
              onClick={[Function]}
              title="Epidemiological Exploration "
            >
              <span
                className="icon-bg"
              >
                <i>
                  EE
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              aria-current={null}
              href="/review"
              onClick={[Function]}
              title="Interactive Analysis"
            >
              <span
                className="icon-bg"
              >
                <i>
                  IA
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs bsd"
          >
            <span
              title="Biological Signature of Diseases"
            >
              <div
                className="dropdown btn-group btn-group-default"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="experiment-dropdown"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  BSD
                </button>
                <ul
                  aria-labelledby="experiment-dropdown"
                  className="dropdown-menu"
                  role="menu"
                >
                  <div
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                    style={
                      Object {
                        "margin": "8px",
                      }
                    }
                  >
                    <span>
                      <p>
                        You have no running experiments.
                      </p>
                      <p>
                        You can start one by selecting a model and configuring an experiment on it.
                      </p>
                    </span>
                  </div>
                </ul>
              </div>
            </span>
          </li>
          <li
            className="uib-dropdown toolbar-icon-bg"
          >
            <a
              aria-current={null}
              href="/profile"
              onClick={[Function]}
              title="Profile"
            >
              <img
                alt="anonymous"
                className="img-circle"
                src="default_user.png"
              />
            </a>
            <ul
              className="dropdown-menu userinfo arrow"
            >
              <li>
                <a
                  href="/profile"
                  ui-sref="profile"
                >
                  <i
                    className="ti ti-user"
                  />
                  <span>
                    Profile
                  </span>
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  href="/login/logout"
                  id="logout-link"
                  ng-click="logout()"
                >
                  <i
                    className="ti ti-shift-right"
                  />
                  <span>
                    Sign out
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs help"
          >
            <div
              className="sc-brqgnP hzzHZ"
            >
              <div
                className="dropdown btn-group btn-group-default"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="help-dropdown"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  help
                   
                  <span
                    className="caret"
                  />
                </button>
                <ul
                  aria-labelledby="help-dropdown"
                  className="dropdown-menu"
                  role="menu"
                >
                  <li
                    className=""
                    role="presentation"
                  >
                    <a
                      href="#"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      <span
                        className="glyphicon-book glyph"
                      />
                       MIP Documentation
                    </a>
                  </li>
                  <li
                    className=""
                    role="presentation"
                  >
                    <a
                      href="#"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      <span
                        className="glyphicon-film glyph"
                      />
                       MIP introduction (video)
                    </a>
                  </li>
                  <div
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                  >
                    <span
                      className="glyphicon-envelope glyph"
                    />
                     Email us at
                     
                    <a
                      href="mailto://support@humanbrainproject.eu"
                    >
                      support@humanbrainproject.eu
                    </a>
                  </div>
                  <div>
                    <div
                      id="feedback-form"
                    />
                  </div>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/hbpapps"
              title="App"
              ui-sref="hbpapps"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-layout-grid3-alt "
                  style={
                    Object {
                      "fontFamily": "themify",
                      "fontWeight": "normal",
                    }
                  }
                />
              </span>
            </a>
          </li>
        </ul>
      </div>
    </header>
  </header>
  <section
    className="main-content sc-fMiknA eEoNzf"
  >
    <div
      className="sc-jAaTju qNdbr"
    >
      <div
        className="sc-jDwBTQ cdxgNu"
      >
        <div
          className="sc-brqgnP hzzHZ"
        >
          <div
            className="dropdown btn-group btn-group-default"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="help-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              help
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="help-dropdown"
              className="dropdown-menu"
              role="menu"
            >
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="glyphicon-book glyph"
                  />
                   MIP Documentation
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="glyphicon-film glyph"
                  />
                   MIP introduction (video)
                </a>
              </li>
              <div
                onKeyDown={[Function]}
                onSelect={[Function]}
              >
                <span
                  className="glyphicon-envelope glyph"
                />
                 Email us at
                 
                <a
                  href="mailto://support@humanbrainproject.eu"
                >
                  support@humanbrainproject.eu
                </a>
              </div>
              <div>
                <div
                  id="feedback-form"
                />
              </div>
            </ul>
          </div>
        </div>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </div>
      <img
        alt="HBP logo"
        src="hbp_logo_320.png"
        title="Human Brain Project"
      />
      <h1>
        THE MEDICAL INFORMATICS PLATFORM
      </h1>
      <div
        className="sc-gPEVay ftfKKm"
      >
        <section>
          <p>
            Thousands of brain images and terabytes of invaluable associated medical data are produced every day at a gigantic pace around the world. The Medical Informatics Platform (MIP) aims to federate this information and provide the tools to the experts to effectively analyse it and advance more rapidly in understanding the neurological and psychiatric diseases. This will in turn allow identifying the biological changes associated and open real possibilities for early diagnosis and personalised medicine.
          </p>
          <p>
            The MIP provides methods to analyse federated data from hospitals, research centres and biobanks. Clinical scientists can develop, share and release results of their research. The MIP aims to bring together people across professional and scientific fields encourages them to actively contribute to the design and development of the services which the MIP provides.
          </p>
        </section>
        <div>
          <section>
            <h3>
              The MIP has three main goals:
            </h3>
            <ul>
              <li>
                Build the tools to federate clinical data, currently inaccessible outside hospital and research archives;
              </li>
              <li>
                Recruit hospitals to contribute to and benefit by using the platform;
              </li>
              <li>
                Develop tools for extracting biological signatures of diseases from multi-level data.
              </li>
            </ul>
          </section>
          <section>
            <h3>
              You can use the MIP
            </h3>
            <ul>
              <li>
                as a clinician, for objective diagnoses and treatment of brain disease;
              </li>
              <li>
                as a neuroscientist, for the application and testing of new models and methods;
              </li>
              <li>
                as a pharmaceutical or biotech company, for disease target discovery.
              </li>
            </ul>
          </section>
        </div>
      </div>
    </div>
  </section>
  <footer
    id="footer"
  >
    <div
      className="clearfix container-fluid"
    >
      <div
        className="row"
      >
        <ul
          className="list-unstyled list-inline col-md-6"
        >
          <li>
            <h6
              className="copyright"
            >
              © 2015-2019
               
              <a
                href="https://www.humanbrainproject.eu/en/"
                title="The Human Brain Project Website"
              >
                Human Brain Project
              </a>
              . All right reserved
            </h6>
          </li>
        </ul>
        <div
          className="col-md-6 text-right"
        >
          
          Mode:
           
          Federation
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`Test App components AppContainer snapshot renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <header
      className="navbar navbar-default navbar-fixed-top "
      id="topnav"
    >
      <div
        className="container-fluid"
      >
        <div
          className="logo-area"
        >
          <a
            aria-current={null}
            className="navbar-brand"
            href="/home"
            onClick={[Function]}
          >
            Human Brain Project
          </a>
        </div>
        <div
          className="app-instance-name"
        >
          <h2>
            MIP
          </h2>
        </div>
        <ul
          className="nav navbar-nav toolbar pull-right"
        >
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              aria-current={null}
              href="/explore"
              onClick={[Function]}
              title="Epidemiological Exploration "
            >
              <span
                className="icon-bg"
              >
                <i>
                  EE
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs"
          >
            <a
              aria-current={null}
              href="/review"
              onClick={[Function]}
              title="Interactive Analysis"
            >
              <span
                className="icon-bg"
              >
                <i>
                  IA
                </i>
              </span>
            </a>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs bsd"
          >
            <span
              title="Biological Signature of Diseases"
            >
              <div
                className="dropdown btn-group btn-group-default"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="experiment-dropdown"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  BSD
                </button>
                <ul
                  aria-labelledby="experiment-dropdown"
                  className="dropdown-menu"
                  role="menu"
                >
                  <div
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                    style={
                      Object {
                        "margin": "8px",
                      }
                    }
                  >
                    <span>
                      <p>
                        You have no running experiments.
                      </p>
                      <p>
                        You can start one by selecting a model and configuring an experiment on it.
                      </p>
                    </span>
                  </div>
                </ul>
              </div>
            </span>
          </li>
          <li
            className="uib-dropdown toolbar-icon-bg"
          >
            <a
              aria-current={null}
              href="/profile"
              onClick={[Function]}
              title="Profile"
            >
              <img
                alt="anonymous"
                className="img-circle"
                src="default_user.png"
              />
            </a>
            <ul
              className="dropdown-menu userinfo arrow"
            >
              <li>
                <a
                  href="/profile"
                  ui-sref="profile"
                >
                  <i
                    className="ti ti-user"
                  />
                  <span>
                    Profile
                  </span>
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  href="/login/logout"
                  id="logout-link"
                  ng-click="logout()"
                >
                  <i
                    className="ti ti-shift-right"
                  />
                  <span>
                    Sign out
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li
            className="toolbar-icon-bg hidden-xs help"
          >
            <div
              className="sc-brqgnP hzzHZ"
            >
              <div
                className="dropdown btn-group btn-group-default"
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="dropdown-toggle btn btn-default"
                  disabled={false}
                  id="help-dropdown"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  type="button"
                >
                  help
                   
                  <span
                    className="caret"
                  />
                </button>
                <ul
                  aria-labelledby="help-dropdown"
                  className="dropdown-menu"
                  role="menu"
                >
                  <li
                    className=""
                    role="presentation"
                  >
                    <a
                      href="#"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      <span
                        className="glyphicon-book glyph"
                      />
                       MIP Documentation
                    </a>
                  </li>
                  <li
                    className=""
                    role="presentation"
                  >
                    <a
                      href="#"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      <span
                        className="glyphicon-film glyph"
                      />
                       MIP introduction (video)
                    </a>
                  </li>
                  <div
                    onKeyDown={[Function]}
                    onSelect={[Function]}
                  >
                    <span
                      className="glyphicon-envelope glyph"
                    />
                     Email us at
                     
                    <a
                      href="mailto://support@humanbrainproject.eu"
                    >
                      support@humanbrainproject.eu
                    </a>
                  </div>
                  <div>
                    <div
                      id="feedback-form"
                    />
                  </div>
                </ul>
              </div>
            </div>
          </li>
          <li
            className="toolbar-trigger toolbar-icon-bg"
          >
            <a
              href="/hbpapps"
              title="App"
              ui-sref="hbpapps"
            >
              <span
                className="icon-bg"
              >
                <i
                  className="ti ti-layout-grid3-alt "
                  style={
                    Object {
                      "fontFamily": "themify",
                      "fontWeight": "normal",
                    }
                  }
                />
              </span>
            </a>
          </li>
        </ul>
      </div>
    </header>
  </header>
  <section
    className="main-content sc-fMiknA eEoNzf"
  >
    <div
      className="sc-jAaTju qNdbr"
    >
      <div
        className="sc-jDwBTQ cdxgNu"
      >
        <div
          className="sc-brqgnP hzzHZ"
        >
          <div
            className="dropdown btn-group btn-group-default"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="help-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              help
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="help-dropdown"
              className="dropdown-menu"
              role="menu"
            >
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="glyphicon-book glyph"
                  />
                   MIP Documentation
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="glyphicon-film glyph"
                  />
                   MIP introduction (video)
                </a>
              </li>
              <div
                onKeyDown={[Function]}
                onSelect={[Function]}
              >
                <span
                  className="glyphicon-envelope glyph"
                />
                 Email us at
                 
                <a
                  href="mailto://support@humanbrainproject.eu"
                >
                  support@humanbrainproject.eu
                </a>
              </div>
              <div>
                <div
                  id="feedback-form"
                />
              </div>
            </ul>
          </div>
        </div>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </div>
      <img
        alt="HBP logo"
        src="hbp_logo_320.png"
        title="Human Brain Project"
      />
      <h1>
        THE MEDICAL INFORMATICS PLATFORM
      </h1>
      <div
        className="sc-gPEVay ftfKKm"
      >
        <section>
          <p>
            Thousands of brain images and terabytes of invaluable associated medical data are produced every day at a gigantic pace around the world. The Medical Informatics Platform (MIP) aims to federate this information and provide the tools to the experts to effectively analyse it and advance more rapidly in understanding the neurological and psychiatric diseases. This will in turn allow identifying the biological changes associated and open real possibilities for early diagnosis and personalised medicine.
          </p>
          <p>
            The MIP provides methods to analyse federated data from hospitals, research centres and biobanks. Clinical scientists can develop, share and release results of their research. The MIP aims to bring together people across professional and scientific fields encourages them to actively contribute to the design and development of the services which the MIP provides.
          </p>
        </section>
        <div>
          <section>
            <h3>
              The MIP has three main goals:
            </h3>
            <ul>
              <li>
                Build the tools to federate clinical data, currently inaccessible outside hospital and research archives;
              </li>
              <li>
                Recruit hospitals to contribute to and benefit by using the platform;
              </li>
              <li>
                Develop tools for extracting biological signatures of diseases from multi-level data.
              </li>
            </ul>
          </section>
          <section>
            <h3>
              You can use the MIP
            </h3>
            <ul>
              <li>
                as a clinician, for objective diagnoses and treatment of brain disease;
              </li>
              <li>
                as a neuroscientist, for the application and testing of new models and methods;
              </li>
              <li>
                as a pharmaceutical or biotech company, for disease target discovery.
              </li>
            </ul>
          </section>
        </div>
      </div>
    </div>
  </section>
  <footer
    id="footer"
  >
    <div
      className="clearfix container-fluid"
    >
      <div
        className="row"
      >
        <ul
          className="list-unstyled list-inline col-md-6"
        >
          <li>
            <h6
              className="copyright"
            >
              © 2015-2019
               
              <a
                href="https://www.humanbrainproject.eu/en/"
                title="The Human Brain Project Website"
              >
                Human Brain Project
              </a>
              . All right reserved
            </h6>
          </li>
        </ul>
        <div
          className="col-md-6 text-right"
        >
          
          Mode:
           
          Federation
        </div>
      </div>
    </div>
  </footer>
</div>
`;
